vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,1202 @@
+---
+title: "Comparing stringr to base R"
+author: "Sara Stoudt"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette aims to compare stringr functions to base R equivalents. Following the groupings of functions from the [stringr cheat sheet](https://github.com/rstudio/cheatsheets/blob/master/strings.pdf), these are the base equivalents to stringr functions.
+
+# Detect Matches
+
+| base | stringr    |
+|------|------------|
+|   `grepl()`   | `str_detect()` |
+|   `grep()`   | `str_which()`  |
+|   `gregexpr()`   | `str_count()`  |
+|  `gregexpr()`    | `str_locate()` |
+
+# Subset Strings
+
+| base | stringr    |
+|------|------------|
+|   `substr()`   | `str_sub()`    |
+|    `grep()`  | `str_subset()` |
+|   `regexpr()` + `regmatches()`   | `str_extract()`|
+|   `regexec()` + `regmatches()`   | `str_match()`  |
+
+
+# Manage Lengths
+
+| base | stringr    |
+|------|------------|
+|   `nchar()`   | `str_length()` |
+|  `sprintf()`    | `str_pad()`    |
+|   `substr()`   | `str_trunc()`  |
+|   `trimws()`   | `str_trim()`   |
+
+# Mutate Strings
+
+| base | stringr        |
+|------|----------------|
+|  `substr()`    | `str_sub()`        |
+|   `sub()`   | `str_replace()`    |
+|   `gsub()`   | `str_replace_all()`|
+|   `tolower()`   | `str_to_lower()`   |
+|   `toupper()`   | `str_to_upper()`   |
+|  `tools::toTitleCase()`    | `str_to_title()`   |
+
+
+# Join and Split
+
+| base | stringr        |
+|------|----------------|
+|   `paste()`   | `str_c()`          |
+|  `replicate()` + `paste()`    | `str_dup()`        |
+|  `strsplit()`    | `str_split_fixed()`|
+|   `paste()` + `eval()`   | `str_glue()`       |
+|    `paste()`  | `str_glue_data()`  |
+
+# Order Strings
+
+| base | stringr        |
+|------|----------------|
+|   `order()`   | `str_order()`      |
+|   `sort()`   | `str_sort()`       |
+
+# Helpers
+
+| base | stringr      |
+|------|--------------|
+|  `iconv()`    | `str_conv()`     |
+|  N/A    | `str_view()`     |
+|   N/A   | `str_view_all()` |
+|  `strwrap()`    | `str_wrap()`     |
+
+```{r}
+library(stringr)
+```
+
+The following examples walk through the examples in the stringr documentation and reproduce the results in base R.
+
+Note: In the following, I use base `apply` functions (rather than tidy [purrr](https://purrr.tidyverse.org/) functions) when illustrating the base alternative to the [stringr](https://stringr.tidyverse.org/) functions. 
+
+## case
+
+*Convert case of a string*
+
+```{r}
+dog <- "The quick brown dog"
+
+## stringr
+str_to_upper(dog)
+str_to_lower(dog)
+str_to_title(dog)
+
+## base
+toupper(dog)
+tolower(dog)
+tools::toTitleCase(dog)
+```
+
+Locale distinctions are not made in base R.
```

Comment:
They are made, they're controlled with global variables (i.e. `Sys.setlocale()`)

Reply:
done in commit 70d0f45